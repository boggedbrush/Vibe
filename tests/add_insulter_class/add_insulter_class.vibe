# VibeSpec: 1.5
patch_type: add_class
file: hello.py
--- code: |
    class Insulter:
        """
        A class that issues a playful insult to a given name.
        """
        def __init__(self, name):
            self.name = name

        def insult(self):
            print(f"{self.name}, you're a nincompoop!")
