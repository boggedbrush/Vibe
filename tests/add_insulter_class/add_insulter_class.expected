# hello.py
def greet(name):
    print(f"Hello, {name}!")


class Greeter:
    def __init__(self, name):
        print(f'Hello {name}')


class Insulter:
    """
    A class that issues a playful insult to a given name.
    """

    def __init__(self, name):
        self.name = name

    def insult(self):
        print(f"{self.name}, you're a nincompoop!")
